src/libraries/Pool.sol

```diff
@@ -317,8 +318,8 @@ library Pool {
         state.feeGrowthGlobalX128 = zeroForOne ? self.feeGrowthGlobal0X128 : self.feeGrowthGlobal1X128;
         state.liquidity = cache.liquidityStart;
 
-        swapFee =
-            cache.protocolFee == 0 ? slot0Start.lpFee : uint24(cache.protocolFee).calculateSwapFee(slot0Start.lpFee);
+        uint24 lpFee = params.fee <= LPFeeLibrary.MAX_LP_FEE ? params.fee : slot0Start.lpFee;
```

Comment:
maybe 1 comment above this just saying 
```
// if the beforeSwap hook returned a valid fee, use that as the LP fee, otherwise load from storage
```
or something?

`Pool.sol` feels quite removed from the hook interaction

Reply:
this line makes me want to do the following:
update `LPFeeLib.validate` to be `isValid returns bool` then other places that call it can check `if(!isValid) revert` and this place can check `if (isValid) params.fee : slot0.lpFee`

this actually follows the model of `ProtocolFeeLibrary` too - currently they behave differently which is weird. Then we should also rename to `ProtocolFeeLibrary.isValid` too hahaali

Reply:
yeah i like the suggestion -- updated for both fee libraries